api/hyperblock/routes.go

```diff
@@ -0,0 +1,64 @@
+package hyperblock
+
+import (
+	"encoding/hex"
+	"net/http"
+
+	apiErrors "github.com/ElrondNetwork/elrond-proxy-go/api/errors"
+	"github.com/ElrondNetwork/elrond-proxy-go/api/shared"
+	"github.com/ElrondNetwork/elrond-proxy-go/data"
+	"github.com/gin-gonic/gin"
+)
+
+// Routes defines the HTTP routes
+func Routes(router *gin.RouterGroup) {
+	router.GET("/by-hash/:hash", ByHashHandler)
+	router.GET("/by-nonce/:nonce", ByNonceHandler)
+}
+
+// ByHashHandler handles "by-hash" requests
+func ByHashHandler(c *gin.Context) {
+	epf, ok := c.MustGet("elrondProxyFacade").(facadeHandler)
+	if !ok {
+		shared.RespondWithInvalidAppContext(c)
+		return
+	}
+
+	hash := c.Param("hash")
+	_, err := hex.DecodeString(hash)
+	if err != nil {
```

Comment:
this check is not needed - it will error on the next check.

Reply:
Exists in the `/block/by-hash` route as well, to prevent bothering observers if the provided `hash` parameter is not a valid hex string.

Reply:
yep. we can.